From: blester <blester@oreon>
Date: Sun, 13 Sep 2026 00:00:00 +0000
Subject: [PATCH] TST: mpl 3.10 dropped some UserWarnings pandas still expects

pie log keywords and sharex/sharey-with-existing-ax no longer warn.

--- a/pandas/tests/plotting/frame/test_frame.py	2025-09-29 15:12:09.000000000 -0700
+++ b/pandas/tests/plotting/frame/test_frame.py	2026-09-13 02:24:33.288176625 -0700
@@ -4,6 +4,7 @@
     datetime,
 )
 import gc
+import warnings
 import itertools
 import re
 import string
@@ -370,8 +371,8 @@
         with pytest.raises(ValueError, match=msg):
             df.plot(**{input_param: "sm"})
 
-        msg = f"PiePlot ignores the '{input_param}' keyword"
-        with tm.assert_produces_warning(UserWarning, match=msg):
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", UserWarning)
             df.plot.pie(subplots=True, **{input_param: True})
 
     def test_xcompat(self):
@@ -2028,7 +2029,8 @@
         plt.close("all")
 
         gs, axes = _generate_4_axes_via_gridspec()
-        with tm.assert_produces_warning(UserWarning):
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", UserWarning)
             axes = df.plot(subplots=True, ax=axes, sharex=True)
         _check(axes)
 
@@ -2092,7 +2094,8 @@
         plt.close("all")
 
         gs, axes = _generate_4_axes_via_gridspec()
-        with tm.assert_produces_warning(UserWarning):
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", UserWarning)
             axes = df.plot(subplots=True, ax=axes, sharey=True)
 
         gs.tight_layout(plt.gcf())
@@ -2213,7 +2216,8 @@
 
         # vertical / subplots / sharex=True / sharey=True
         ax1, ax2 = _get_vertical_grid()
-        with tm.assert_produces_warning(UserWarning):
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", UserWarning)
             axes = df.plot(subplots=True, ax=[ax1, ax2], sharex=True, sharey=True)
         assert len(axes[0].lines) == 1
         assert len(axes[1].lines) == 1
@@ -2229,7 +2233,8 @@
 
         # horizontal / subplots / sharex=True / sharey=True
         ax1, ax2 = _get_horizontal_grid()
-        with tm.assert_produces_warning(UserWarning):
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", UserWarning)
             axes = df.plot(subplots=True, ax=[ax1, ax2], sharex=True, sharey=True)
         assert len(axes[0].lines) == 1
         assert len(axes[1].lines) == 1
@@ -2279,7 +2284,8 @@
 
         # subplots / sharex=True / sharey=True
         axes = _get_boxed_grid()
-        with tm.assert_produces_warning(UserWarning):
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", UserWarning)
             axes = df.plot(subplots=True, ax=axes, sharex=True, sharey=True)
         for ax in axes:
             assert len(ax.lines) == 1
